src/main/java/mate/academy/bookshop/service/BookServiceImpl.java

```diff
@@ -1,27 +1,41 @@
 package mate.academy.bookshop.service;
 
 import java.util.List;
+import java.util.Random;
+import lombok.RequiredArgsConstructor;
+import mate.academy.bookshop.dto.BookDto;
+import mate.academy.bookshop.dto.CreateBookRequestDto;
+import mate.academy.bookshop.exception.EntityNotFoundException;
+import mate.academy.bookshop.mapper.BookMapper;
 import mate.academy.bookshop.model.Book;
 import mate.academy.bookshop.repository.BookRepository;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
+@RequiredArgsConstructor
 @Service
 public class BookServiceImpl implements BookService {
     private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
 
-    @Autowired
-    public BookServiceImpl(BookRepository bookRepository) {
-        this.bookRepository = bookRepository;
+    @Override
+    public BookDto save(CreateBookRequestDto requestDto) {
+        Book book = bookMapper.toModel(requestDto);
+        book.setIsbn("" + new Random().nextInt(1000));
```

Comment:
Why is it generated? I'm not sure that stores generate ISBN of some book they want to sell. So, make it an API parameter.